### ISSUE TITLE
ECR put_image does not appear to be using the passed imageDigest when retrieved via describe_images

### ISSUE THREAD

**Issue by cloudwitch** [getmoto/moto#6997]
ECR `put_image` does not appear to be using the passed `imageDigest` when retrieved via `describe_images`.

I'm trying to write unit tests to make sure the manifest of a multi-arch image contains the child images, but when I do a `describe_images` on any of the images, the `imageDigest` is different.

It appears the `put_image` is disregarding the `imageDigest` argument and using a randomly generated one. This makes it impossible to make sure what is uploaded to the mock ECR is actually there.

The issue seems to be for tagged and untagged images. However, if it's untagged, you can get it to return the same `imageDigest` on multiple runs, but this still does not match the passed `imageDigest`.

Repro:

Testing in a Python 3.11 container, `moto==4.2.7`, `boto3==1.28.79`.

Save this:

```python
#!/usr/bin/env python3

import os
import json
import boto3
from moto import mock_ecr

os.environ["AWS_REGION"] = "us-east-1"
os.environ["AWS_ACCESS_KEY_ID"] = "testing"
os.environ["AWS_SECRET_ACCESS_KEY"] = "testing"
os.environ["AWS_SECURITY_TOKEN"] = "testing"
os.environ["AWS_SESSION_TOKEN"] = "testing"
os.environ["AWS_DEFAULT_REGION"] = "us-east-1"
os.environ["ACCOUNT_NUMBER"] = "123456789012"


def put_tagged_image(ecr_client, repo_name, image_manifest, image_manifest_media_type, image_tag, image_digest):
    response = ecr_client.put_image(
        repositoryName=repo_name,
        imageManifest=json.dumps(image_manifest),
        imageManifestMediaType=image_manifest_media_type,
        imageTag=image_tag,
        imageDigest=image_digest,
    )
    return response


def put_untagged_image(ecr_client, repo_name, image_manifest, image_manifest_media_type, image_digest):
    response = ecr_client.put_image(
        repositoryName=repo_name,
        imageManifest=json.dumps(image_manifest),
        imageManifestMediaType=image_manifest_media_type,
        imageDigest=image_digest,
    )
    return response


if __name__ == "__main__":
    with mock_ecr():
        mock_ecr_client = boto3.client("ecr")
        mock_ecr_client.create_repository(repositoryName="test-repo-1")
        test_manifest = {
            "manifests": [
                {
                    "digest": "sha256:181210f8f9c779c26da1d9b2075bde0127302ee0e3fca38c9a83f5b1dd8e5d3b",
                    "mediaType": "application/vnd.oci.image.manifest.v1+json",
                    "platform": {"architecture": "amd64", "os": "linux"},
                    "size": 2010,
                },
                {
                    "digest": "sha256:cdfba543ee8ef7fdb3d8b587648cc22dd792bbd6272cc5447307c7c106c2374c",
                    "mediaType": "application/vnd.oci.image.manifest.v1+json",
                    "platform": {"architecture": "arm64", "os": "linux"},
                    "size": 2010,
                },
                {
                    "annotations": {
                        "vnd.docker.reference.digest": "sha256:181210f8f9c779c26da1d9b2075bde0127302ee0e3fca38c9a83f5b1dd8e5d3b",
                        "vnd.docker.reference.type": "attestation-manifest",
                    },
                    "digest": "sha256:d928266e89b1da2263838e86df3a430548ca1768bedbc9d4b20f9e370d5518df",
                    "mediaType": "application/vnd.oci.image.manifest.v1+json",
                    "platform": {"architecture": "unknown", "os": "unknown"},
                    "size": 566,
                },
                {
                    "annotations": {
                        "vnd.docker.reference.digest": "sha256:cdfba543ee8ef7fdb3d8b587648cc22dd792bbd6272cc5447307c7c106c2374c",
                        "vnd.docker.reference.type": "attestation-manifest",
                    },
                    "digest": "sha256:15e12e46851fa9ef6a0ad3b3b53e513de6d684a173f0199a20456795718a7f31",
                    "mediaType": "application/vnd.oci.image.manifest.v1+json",
                    "platform": {"architecture": "unknown", "os": "unknown"},
                    "size": 566,
                },
            ],
            "mediaType": "application/vnd.oci.image.index.v1+json",
            "schemaVersion": 2,
        }
        response_1 = put_tagged_image(
            ecr_client=mock_ecr_client,
            repo_name="test-repo-1",
            image_manifest_media_type="application/vnd.oci.image.manifest.v1+json",
            image_manifest=test_manifest,
            image_tag="test1",
            image_digest="sha256:c2956073ca4d44e089b39ea86a94619ff4929917ddf060816f673974d3acd7a4",
        )
        response_2 = put_untagged_image(
            ecr_client=mock_ecr_client,
            repo_name="test-repo-1",
            image_manifest_media_type="application/vnd.oci.image.manifest.v1+json",
            image_manifest={
                "config": {
                    "digest": "sha256:6442bc26a7c562f5afe6467dab36365c709909f6a81afcecfc0c25cff0f1bab0",
                    "mediaType": "application/vnd.docker.container.image.v1+json",
                    "size": 5205,
                },
                "layers": [
                    {
                        "digest": "sha256:b35e87b5838011a3637be660e4238af9a55e4edc74404c990f7a558e7f416658",
                        "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
                        "size": 26690191,
                    }
                ],
                "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
                "schemaVersion": 2,
            },
            image_digest="sha256:826b6832e45ba17d625debc95ae8554e148550b00c05b47fa8f7be1c555bc83c",
        )
        print(f"test1 is: {response_1['image']['imageId']['imageDigest']}")
        print(f"test2 is: {response_2['image']['imageId']['imageDigest']}")

```

Then run it multiple times, you'll see the `imageDigest` is not the same.

```bash
root@python3.11container:/# ./test.py 
test1 is: sha256:bdd2d3af3a5a1213497d4f1f7bfcda898274fe9cb5401bbc0190885664708fc2
test2 is: sha256:9ebe3187b224888b53c1b429e2b8f23eb57168825fea06f7d352beac25a57116
root@python3.11container:/# ./test.py 
test1 is: sha256:8241649609f88ccd2a0a5b233a07a538ec313ff6adf695aa44a969dbca39f67d
test2 is: sha256:9ebe3187b224888b53c1b429e2b8f23eb57168825fea06f7d352beac25a57116
```

**Comment by rafcio19** [comment]
@bblommers I can take a look

**Comment by cloudwitch** [comment]
Thank you so much fafcio19!

**Comment by bblommers** [comment]
Welcome to Moto, @cloudwitch!

This should be fixed as of moto >= 4.2.8.dev25 - please let us know if you run into any other issues.

**Comment by cloudwitch** [comment]
It's fixed! Thank you!!!

```
root@python3.11container:/# ./test.py 
test1 is: sha256:c2956073ca4d44e089b39ea86a94619ff4929917ddf060816f673974d3acd7a4
test2 is: sha256:826b6832e45ba17d625debc95ae8554e148550b00c05b47fa8f7be1c555bc83c
root@python3.11container:/# ./test.py 
test1 is: sha256:c2956073ca4d44e089b39ea86a94619ff4929917ddf060816f673974d3acd7a4
test2 is: sha256:826b6832e45ba17d625debc95ae8554e148550b00c05b47fa8f7be1c555bc83c
```
